x/btcstaking/types/create_delegation_parser.go

```diff
@@ -0,0 +1,268 @@
+package types
+
+import (
+	"fmt"
+	math "math"
+
+	"github.com/babylonlabs-io/babylon/btcstaking"
+	bbn "github.com/babylonlabs-io/babylon/types"
+	btcckpttypes "github.com/babylonlabs-io/babylon/x/btccheckpoint/types"
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/wire"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type ParsedPublicKey struct {
+	PublicKey *btcec.PublicKey
+	BbnPk     *bbn.BIP340PubKey
+}
+
+func NewParsedPublicKey(key *bbn.BIP340PubKey) (*ParsedPublicKey, error) {
+	if key == nil {
+		return nil, fmt.Errorf("cannont parse nil *bbn.BIP340PubKey")
+	}
+	pk, err := key.ToBTCPK()
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to parer *bbn.BIP340PubKey: %w", err)
+	}
+
+	return &ParsedPublicKey{
+		PublicKey: pk,
+		BbnPk:     key,
+	}, nil
+}
+
+type ParsedBIP340Signature struct {
+	Signature *schnorr.Signature
+	BbnSig    *bbn.BIP340Signature
```

Comment:
```suggestion
	*schnorr.Signature
	*bbn.BIP340Signature
```
Same here. `BbnSig` is a bit confusing